x/bank/keeper/send.go

```diff
@@ -260,14 +260,12 @@ func (k BaseSendKeeper) SendCoins(ctx context.Context, fromAddr, toAddr sdk.AccA
 		return err
 	}
 
-	// Unreleased improvement in v0.51
-	// https://github.com/cosmos/cosmos-sdk/pull/20517
-	toAddr, err = k.sendRestriction.apply(ctx, fromAddr, toAddr, amt)
+	err = k.subUnlockedCoins(ctx, fromAddr, amt)
```

Comment:
match order in https://github.com/cosmos/cosmos-sdk/blob/v0.50.12/x/bank/keeper/send.go#L211

it's safe to deduct first. also blacklist is a feature for runTx whose cache is discarded on error. blacklist is bypassed in begin/end blockers.